ydb/core/cms/console/console_handshake.cpp

```diff
@@ -0,0 +1,104 @@
+#include "defs.h"
+
+#include "configs_config.h"
+#include "console_impl.h"
+#include "console_configs_manager.h"
+#include <ydb/public/api/grpc/draft/ydb_dynamic_config_v1.grpc.pb.h>
+
+#include <ydb/core/base/tablet_pipe.h>
+
+#include <ydb/library/actors/core/hfunc.h>
+
+namespace NKikimr::NConsole {
+
+void TConfigsManager::Handle(TEvBlobStorage::TEvControllerProposeConfigRequest::TPtr &ev, const TActorContext &ctx) {
+    const auto& record = ev->Get()->Record;
+    const auto& proposedConfigHash = record.GetConfigHash();
+    const auto& proposedConfigVersion = record.GetConfigVersion();
+
+    auto currentConfigHash = std::hash<std::string>{}(YamlConfig);
+    if (Self.CurrentSenderId != ev->Sender) {
+        auto endSessionEvent = std::make_unique<IEventHandle>(Self.SelfId(), Self.CurrentSenderId, new TEvConsole::TEvCloseSession());
+        if (Self.CurrentInterconnectSession) {
+            endSessionEvent->Rewrite(TEvInterconnect::EvForward, Self.CurrentInterconnectSession);
+            ctx.Send(endSessionEvent.release());
+        }
+        else {
+            return;
+        }
+    }
+    Self.CurrentSenderId = ev->Sender;
+    Self.CurrentPipeServerId = ev->Recipient;
+    Self.CurrentInterconnectSession = ev->InterconnectSession;
+    auto response = std::make_unique<TEvBlobStorage::TEvControllerProposeConfigResponse>();
+    auto& responseRecord = response->Record;
+
+    if (YamlVersion > record.GetConfigVersion()) {
+        responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::UnexpectedConfig);
+        responseRecord.SetProposedConfigVersion(proposedConfigVersion);
+        responseRecord.SetConsoleConfigVersion(YamlVersion);
+        SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+        LOG_ALERT_S(ctx, NKikimrServices::CMS, "Unexpected proposed config.");
+        return;
+    }
+    if (YamlVersion == record.GetConfigVersion()) {
+        responseRecord.SetStatus(NKikimrBlobStorage::TEvControllerProposeConfigResponse::CommitIsNotNeeded);
+        SendInReply(ev->Sender, Self.CurrentInterconnectSession, std::move(response));
+        return;
+    }
+    if (proposedConfigHash != currentConfigHash) {
```

Comment:
yep, you have older version than proposed one; it's quite obvious hashes will differ :)
we check 'em only when versions are the same